styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -161,13 +161,14 @@ public QueuedStateManager(
   private void initialize(WorkflowInstance workflowInstance) {
     // Write active state to datastore
 
-    final long counter;
+    final long nextCounter;
     try {
-      counter = storage.getLatestStoredCounter(workflowInstance).orElse(NO_EVENTS_PROCESSED);
+      final long counter = storage.getLatestStoredCounter(workflowInstance).orElse(NO_EVENTS_PROCESSED);
+      nextCounter = counter + 1;
```

Comment:
This also means that the NEW RunState will have counter 0 instead of -1 in case of the first trigger, right? I don't see any problem with that (it probably makes more sense even)

Reply:
yeah, don't block on my nitpick :)